src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -784,6 +792,47 @@ void ShenandoahHeap::op_uncommit(double shrink_before, size_t shrink_until) {
   }
 }
 
+void ShenandoahHeap::service_uncommit(double shrink_before, size_t shrink_until) {
```

Comment:
I propose we call it `maybe_uncommit`. I think you can then also inline `entry_uncommit` right here, as we don't need the interface between control thread and heap anymore.